### PR TITLE
PHP 8.2 updates

### PR DESCRIPTION
**Pull request by cviebrock**

### CODE REVIEW

File: tests/demo.php [hunk 1]
```diff
@@ -1,11 +1,9 @@
 <?php
 
 /**
- * @package   XML_RPCAjax
- t* @copyright 2005-2016 silverorange
+ * t* @copyright 2005-2016 silverorange
```
Comment (gauthierm) [inline tests/demo.php:4]: this is a typo

File: www/javascript/xml-rpc-ajax.js [hunk 2]
```diff
@@ -434,31 +430,33 @@ XML_RPC_Request.getXmlRpcValue = function(value, type)
...
+    case 'string':
+      return 'string';
+    case 'boolean':
+      return 'boolean';
+    case 'number':
+      // this is a best guess
+      return 'double';
+    // both array and objects are 'object'
+    case 'object':
+      if (value instanceof Array) return 'array';
```
Comment (gauthierm) [inline www/javascript/xml-rpc-ajax.js:444]: Can these get braces?

Comment (cviebrock) [reply www/javascript/xml-rpc-ajax.js:444]: I'm surprised prettier didn't change this ... manually fixing.

Comment (cviebrock) [reply www/javascript/xml-rpc-ajax.js:444]: (I guess it would be an eslint thing, but we're not using that.)

**Pull request merged**